eth/beacon/state_machines/base.py

```diff
@@ -83,25 +83,30 @@ def get_block_class(cls) -> Type[BaseBeaconBlock]:
             return cls.block_class
 
     @classmethod
-    @functools.lru_cache(maxsize=32)
     @to_tuple
-    def get_prev_blocks_to_slot(cls,
-                                last_block_hash: Hash32,
-                                chaindb: BaseBeaconChainDB,
-                                depth: int,
-                                start_slot: int) -> Iterable[BaseBeaconBlock]:
+    def get_prev_blocks(cls,
+                        last_block_hash: Hash32,
+                        chaindb: BaseBeaconChainDB,
+                        searching_depth: int,
```

Comment:
*nitpick*

I think `search_depth` is a more natural name for this.  And maybe even `max_search_depth` since this seems to be an upper bound.